### PR TITLE
Rescale images used in Input

### LINKED ISSUE #841: When taking photos allow for setting size + quality
As we now have phone cameras with almost 50 mega pixel resolution, it is super important to have the option to minimize file-sizes (No desire to synch up gigabyte-sized Input-projects with QGIS). As a work-around I am using the OSS project [Open Camera](https://opencamera.org.uk/). Nonetheless, it would be cool if Input had some support for keeping image files small.

### PR DESCRIPTION
**Pull request by alexbruy**
Resize/rescale images selected for using in Input according to the Mergin project setting (defined in the QGIS plugin). Keep EXIF data of the original file.

Setting image quality done in QGIS plugin, related pull-request lutraconsulting/qgis-mergin-plugin/pull/383

Fixes #841.

### CODE REVIEW

**Comment by alexbruy** [comment]
Seems we should upgrade exiv2 to 1.0.0 or build with C++11/C++14 compiler as it uses `auto_ptr`.

File: core/imageutils.cpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  QString newPath = QStringLiteral( "%1/%2_rescaled.%3" ).arg( fi.path(), fi.baseName(), fi.completeSuffix() );
+
+  if ( !rescaledImage.save( newPath ) )
+  {
+    CoreUtils::log( "rescaling image", QStringLiteral( "Failed to save rescaled image" ) );
+    return false;
+  }
+
+  // copy EXIF from source image to rescaled image
+  copyExifMetadata( path, newPath );
```
Comment (tomasMizera) [inline core/imageutils.cpp:118]: Shall we continue if `copyExifMetadata` fails to copy the EXIF data?

File: app/inpututils.cpp [hunk 2]
```diff
@@ -1625,3 +1626,10 @@ QString InputUtils::iconFromGeometry( const QgsWkbTypes::GeometryType &geometry
     default: return QString( "qrc:/mIconTableLayer.svg" );
   }
 }
+
+bool InputUtils::rescaleImage( const QString &path, QgsProject *activeProject )
+{
+  qDebug() << "rescaleImage called";
```
Comment (tomasMizera) [inline app/inpututils.cpp:1632]: This seems like forgotten debug log

File: app/qml/ExternalResourceBundle.qml [hunk 3]
```diff
@@ -97,6 +97,11 @@ Item {
          */
         property var confirmImage: function confirmImage(itemWidget, prefixToRelativePath, value) {
           if (value) {
+            console.log("TRY TO RESCALE" + value)
```
Comment (tomasMizera) [inline app/qml/ExternalResourceBundle.qml:100]: This seems like forgotten debug log

File: app/qml/ExternalResourceBundle.qml [hunk 4]
```diff
@@ -97,6 +97,11 @@ Item {
...
+            if (!__inputUtils.rescaleImage(value, __activeProject.qgsProject))
+            {
+              console.log("RESCALE FAILED" + value)
+            }
             var newCurrentValue = __inputUtils.getRelativePath(value, prefixToRelativePath)
             itemWidget.editorValueChanged(newCurrentValue, newCurrentValue === "" || newCurrentValue === null)
```
Comment (tomasMizera) [inline app/qml/ExternalResourceBundle.qml:106]: No need for the `console.log`, it is not included in logs from clients and `rescaleImage` already logs when it fails
```suggestion
            if (__inputUtils.rescaleImage(value, __activeProject.qgsProject))
            {
              var newCurrentValue = __inputUtils.getRelativePath(value, prefixToRelativePath)
              itemWidget.editorValueChanged(newCurrentValue, newCurrentValue === "" || newCurrentValue === null)
            }
```

**Review by tomasMizera — COMMENTED** [review]
This is looking really good, good job @alexbruy! 🎉

**Comment by PeterPetrik** [comment]
> Seems we should upgrade exiv2 to 1.0.0 or build with C++11/C++14 compiler as it uses `auto_ptr`.

Is it about building Input with C++11?  I think it would be good to bump it to C++11/14/17 as QGIS should be on it anyways https://github.com/qgis/QGIS/blob/799679cfe4ddcc16880e46833e748754b4244949/CMakeLists.txt#L688

@wonder-sk ?

**Comment by PeterPetrik** [comment]
SDK issue: https://github.com/lutraconsulting/input-sdk/issues/68

**Review by PeterPetrik — APPROVED** [review]

**Pull request merged**